src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -2170,6 +2212,13 @@ std::error_code Session::receive_unbound_message()
 }
 
 
+std::error_code Session::receive_query_error_message(int error_code, std::string_view message, int64_t query_version)
+{
+    logger.info("Received QUERY_ERROR \"%1\" (error_code=%2, query_verison=%3)", message, error_code, query_version);
+    on_flx_sync_error(query_version, std::string_view(message.data(), message.size())); // throws
+    return {};
```

Comment:
Are you planning to conditionally return some error in the future?

Reply:
hmm, i don't think so? all the other `receive_*_message` functions return an error code, so I made this return one just so they all behave the same. it doesn't have to be this way though if it feels messy or wrong to you.

Reply:
ok that's fine